CHANGELOG.md

```diff
@@ -1,16 +1,69 @@
 ## 0.17.11 (Unreleased)
 
+### Major Changes
+
+#### Secrets and Pluggable Encryption
+
+- The Pulumi engine and Python and NodeJS SDKs now have support for tracking values as "secret" to ensure they are
+  encrypted when being persisted in a state file.
+
+  Any existing value may be turned into a secret by calling `pulumi.secret(<value>)` (NodeJS) or
+  `Output.secret(<value>`) (Python).  In both cases, the returned value is an Output which may be passed around
+  like any other.  If this value flows into a resource, the plaintext will not be stored in the state file, but instead
+  It will be encrypted, just like values added to config with `pulumi config set --secret`.
+
+  If an output which has been marked as secret is combiend with other outputs (either via `all` or `apply`) the
+  resulting output value will also be treated as a secret.
+
+  You can verify that values are being stored as you expect by running `pulumi stack export`, When values are encrypted
+  in the state file, they appear as an object with a special signiture key and a ciphertext property.
+
+  When ouputs of a stack are secrets, `pulumi stack output` will show `[secret]` as the value, by default.  You can
+  pass `--show-secrets` to `pulumi stack output` in order to see the actual raw value.
+
+  **Known Issues**
```

Comment:
I would not include this in the changleog. We can leave this in the PR or issue linked from the changelog.